packages/lab/src/date-picker/DatePicker.tsx

```diff
@@ -21,11 +24,19 @@ export interface DatePickerBaseProps {
   children?: ReactNode;
   /** the open/close state of the overlay. The open/close state will be controlled when this prop is provided. */
   open?: boolean;
+  /** When `open` is uncontrolled, set this to `true` to open on click */
+  openOnClick?: boolean;
   /**
    * Handler for when open state changes
    * @param newOpen - true when opened
+   * @param event - event that triggered the state change
+   * @param reason - reason for the the state change
    */
-  onOpen?: (newOpen: boolean) => void;
+  onOpen?: (
```

Comment:
```suggestion
  onOpenChange?: (
```

Would align with other components?

Reply:
good shout, renamed